testutils/infra/smtpd_mock.py

```diff
@@ -76,8 +93,80 @@ def assert_called(self, email):
         assert len(m.data) > 0
 
 
+class SMTPGmail:
+    def __init__(self, server, address, password):
+        self._server = server
+        self._address = address
+        self._password = password
+
+    def connect(self):
+        mail = imaplib.IMAP4_SSL(self._server)
+        mail.login(self._address, self._password)
+        mail.select("INBOX")
+        return mail
+
+    def await_messages(self, n=1, timeout=None):
+        timeout = int(timeout) if timeout else 60
+        for i in range(timeout):
+            time.sleep(1)
+            if len(self.messages()) >= n:
+                break
+
+    def messages(self):
+        mail = self.connect()
+        try:
+            type, data = mail.search(None, "ALL")
+            assert type == "OK"
+            messages = []
+            mail_ids = data[0].decode("utf-8").split()
+            if not mail_ids:
+                return messages
+            for i in range(int(mail_ids[-1]), int(mail_ids[0]) - 1, -1):
+                type, data = mail.fetch(str(i), "(RFC822)")
+                body = data[0][1]
+                msg = email.message_from_string(body.decode("utf-8"))
+                messages.append(
+                    Message(None, msg["from"], [msg["to"]], msg["subject"], body)
+                )
+            return messages
+        finally:
+            mail.close()
+
+    def filtered_messages(self, email):
+        messages = self.messages()
+        return tuple(filter(lambda m: email in m.rcpttos[0], messages))
+
+    def assert_called(self, email):
+        msgs = self.filtered_messages(email)
+        assert len(msgs) == 1
+        m = msgs[0]
+        assert "@mender.io" in m.mailfrom, m.mailfrom
+        assert email in m.rcpttos[0]
+        assert len(m.data) > 0
+
+    def clear(self):
+        mail = self.connect()
+        try:
+            type, data = mail.search(None, "ALL")
+            assert type == "OK"
+            mail_ids = data[0].decode("utf-8").split()
+            if not mail_ids:
+                return
+            for i in range(int(mail_ids[-1]), int(mail_ids[0]) - 1, -1):
+                mail.store(str(i), "+FLAGS", "\\Deleted")
+            mail.expunge()
+        finally:
+            mail.close()
+
+
 @pytest.fixture(scope="function")
 def smtp_mock():
+    global GMAIL_SERVER, GMAIL_ADDRESS, GMAIL_PASSWORD
```

Comment:
Nitpick: using global variables is considered a bad practice - is this really necessary?